operations/mimir/store-gateway-autoscaling.libsonnet

```
@@ -0,0 +1,70 @@
+{
+  // Allow store-gateways to be safely downscaled using the "prepare-downscale" and "downscale-leader" features
+  // of the rollout-operator. Only available when using multi-zone store-gateways. This feature can be enabled
+  // via the store_gateway_automated_downscale_enabled flag. When enabled, the number of replicas in zone B will
+  // be scaled to match the number of replicas in zone A on a configurable delay. Correspondingly, zone C will
+  // follow zone B on the same configurable delay. The default delay can be changed via
+  // $._config.store_gateway_autoscaling_min_time_between_zones_downscale. The number of replicas in zone A
+  // is still controlled by $._config.multi_zone_store_gateway_replicas until a reasonable metric to scale it
+  // automatically is determined and tested.
+
+  _config+: {
+    store_gateway_automated_downscale_enabled: true,
+    // Give more time if lazy-loading is disabled.
+    store_gateway_autoscaling_min_time_between_zones_downscale: if $._config.store_gateway_lazy_loading_enabled then '15m' else '60m',
```

[nit] I would rename to match the "enabled" config option name.

```suggestion
    store_gateway_automated_downscale_min_time_between_zones: if $._config.store_gateway_lazy_loading_enabled then '15m' else '60m',
```